qiskit/quantum_info/operators/mixins/group.py

```diff
@@ -0,0 +1,194 @@
+        return self._tensor(other, self)
+
+    def compose(self, other, qargs=None, front=False):
+        """Return the operator composition with another {cls}.
+
+        Args:
+            other ({cls}): a {cls} object.
+            qargs (list or None): Optional, a list of subsystem positions to
+                                  apply other on. If None apply on all
+                                  subsystems (default: None).
+            front (bool): If True compose using right operator multiplication,
+                          instead of left multiplication [default: False].
+
+        Returns:
+            {cls}: The composed {cls}.
+
+        Raises:
+            QiskitError: if other cannot be converted to an operator, or has
+                         incompatible dimensions for specified subsystems.
+
+        .. note::
+            Composition (``@``) is defined as `left` matrix multiplication for
+            matrix operators. That is that ``A @ B`` is equal to ``B * A``.
+            Setting ``front=True`` returns `right` matrix multiplication
+            ``A * B`` and is equivalent to the :meth:`dot` method.
+        """.format(cls=type(self).__name__)
+        if qargs is None:
+            qargs = getattr(other, 'qargs', None)
+        return self._compose(other, qargs=qargs, front=front)
+
+    def dot(self, other, qargs=None):
+        """Return the right multiplied operator self * other.
+
+        Args:
+            other ({cls}): an operator object.
+            qargs (list or None): Optional, a list of subsystem positions to
+                                  apply other on. If None apply on all
+                                  subsystems (default: None).
+
+        Returns:
+            {cls}: The operator self * other.
+
+        .. note::
+            The dot product can be obtained using the ``*`` binary operator.
+            Hence ``a.dot(b)`` is equivalent to ``a * b``. Left operator
+            multiplication can be obtained using the :meth:`compose` method.
+
+        """.format(cls=type(self).__name__)
+        return self.compose(other, qargs=qargs, front=True)
+
+    def power(self, n):
+        """Return the compose of a operator with itself n times.
+
+        Args:
+            n (int): the number of times to compose with self (n>0).
+
+        Returns:
+            {cls}: the n-times composed operator.
+
+        Raises:
+            QiskitError: if the input and output dimensions of the operator
+                         are not equal, or the power is not a positive integer.
+        """.format(cls=type(self).__name__)
+        # NOTE: if a subclass can have negative or non-integer powers
+        # this method should be overridden in that class.
+        if not isinstance(n, Integral) or n < 1:
+            raise QiskitError("Can only power with positive integer powers.")
+        ret = self
+        for _ in range(1, n):
+            ret = ret.dot(self)
+        return ret
+
+    @classmethod
```

Comment:
Why is `_tensor` a class method (while the others are ordinary methods)?

Reply:
This was so you can write both `tensor` or `expand` in terms of it, otherwise you need a separate abstract function for each.

Reply:
I ended up changing back to tensor/expand being abstract in mixins, though I left `_tensor_ as the actual implementation in most classes.